src/providers/tpm/key_attestation.rs

```diff
@@ -141,11 +143,32 @@ impl Provider {
             )
             .map_err(|e| {
                 format_error!("Failed to activate credential", e);
-                utils::to_response_status(e)
+                key_attest_response_status(e)
             })?;
 
         Ok(attest_key::Result::ActivateCredential {
             credential: credential.into(),
         })
     }
 }
+
+fn key_attest_response_status(error: Error) -> ResponseStatus {
+    match error {
+        Error::Tss2Error(e) => match e.kind() {
+            Some(Tss2ResponseCodeKind::BadAuth) => {
+                error!("Wrong authentication value for attesting key");
+                ResponseStatus::PsaErrorGenericError
```

Comment:
no test for this type of error?

Reply:
Will add one, have to do it in a config test, though, because it requires the `endorsement_auth` to be wrong

Reply:
Ok, I've added a test for this